### PR TITLE
mirror: Delete tags in mirror which are removed for original repo.

### LINKED ISSUE #5192: Unable to delete tags when syncing mirror
- Gitea version (or commit ref): 1.6.0-rc1 / aeb5655
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:
```
[...odels/repo_mirror.go:335 SyncMirrors()] [E] Failed to synchronize tags to releases for repository: GetTagCommitID: object does not exist [id: refs/tags/test, rel_path: ]
```

## Description
Step to reproduce:
1. Delete a tag from `Repo1`
2. Sync `Repo2` (mirror of `Repo1`)

https://github.com/go-gitea/gitea/blob/aeb5655c25053bdcd7eee94ea37df88468374162/models/release.go#L488-L495

### PR DESCRIPTION
**Pull request by HarshitOnGitHub**
This bug was being caused by an error in the logic in `release.go`.
Credit to @yasuokav for tracing the root of the issue.
I have tested the changes manually as well as added an automated test.

Fixes: #5192.

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
@HarshitOnGitHub Thanks for another PR 😄 Best wishes for the new year 🎉

**Comment by HarshitOnGitHub** [comment]
>Best wishes for the new year :tada:

Same to you @techknowlogick and to the whole team of gitea. :)

File: models/release.go [hunk 1]
```diff
@@ -479,10 +479,10 @@ func SyncReleasesWithTags(repo *Repository, gitRepo *git.Repository) error {
 				continue
 			}
 			commitID, err := gitRepo.GetTagCommitID(rel.TagName)
-			if err != nil {
+			if err != nil && !git.IsErrNotExist(err) {
 				return fmt.Errorf("GetTagCommitID: %v", err)
 			}
-			if !gitRepo.IsTagExist(rel.TagName) || commitID != rel.Sha1 {
+			if git.IsErrNotExist(err) || commitID != rel.Sha1 {
```
Comment (lunny) [inline models/release.go:485]: These two condition are different. If `git.IsErrNotExist(err)`, it should be delted. But for `commitID != rel.Sha`, the tag should be retag but not delete.

Comment (lunny) [inline models/release.go:485]: These two condition are different. If `git.IsErrNotExist(err)`, it should be delted. But for `commitID != rel.Sha`, the tag should be retag but not delete.

**Comment by HorlogeSkynet** [comment]
Up, could we have a feedback or a patch for #6237 from you @HarshitOnGitHub ? Bye :wave:

**Pull request merged**